### PR TITLE
Fix session user data not update properly after calling /profile update routes and updated in MongoDB

### PR DESCRIPTION
**Pull request by nguyenit67**
solves #584  and #575

### CODE REVIEW

File: src/db/repository/UserRepository.ts [hunk 1]
```diff
@@ -23,6 +23,18 @@ export default class UserRepository {
 		}
 	}
 
+	async getUserAndUpdateById(id: string, updateParams: Partial<User>) {
+		try {
+			const updatedUser = await this.userModel
+				.findOneAndUpdate({ _id: id }, { $set: updateParams }, { new: true })
+				.lean()
+				.exec();
+			return updatedUser;
```
Comment (Enubia) [inline src/db/repository/UserRepository.ts:32]: You can just return the update statement, no need to assign the result first

**Review by Enubia — CHANGES_REQUESTED** [review]

File: src/db/repository/UserRepository.ts (hunk 1 above)
Comment (nick-w-nick) [inline src/db/repository/UserRepository.ts:32]:
```suggestion
			return await this.userModel
				.findOneAndUpdate({ _id: id }, { $set: updateParams }, { new: true })
				.lean()
				.exec();
```

@nguyenit67 here's a suggestion you can just apply to match @Enubia's change request so you don't have to do it manually.

Comment (Enubia) [reply src/db/repository/UserRepository.ts:32]: Thanks for the change suggestion, couldn't figure out how to do that in the mobile app haha

Comment (nguyenit67) [reply src/db/repository/UserRepository.ts:32]: Ok I think that will make the code more concise and aligned with existed codebase

Comment (nguyenit67) [reply src/db/repository/UserRepository.ts]: @nick-w-nick thanks pal!

Comment (nguyenit67) [reply src/db/repository/UserRepository.ts]: Also, what's your opinion about assigning return value to an easy to understand name variable, for further debug/logging/doing more stuff using that variable? I want to learn more about coding convention/style guide from an experienced dev like you, thanks pretty much

**Review by Enubia — APPROVED** [review]

**Pull request merged**